### DISCUSSION TITLE
API for annotation queues to programmatically manage them (enqueue, dequeue, status)

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#4714] [Ideas]
### Describe the feature or potential improvement

Let's track interest for an API for annotation queues. Currently this is an UI-only feature.

**Comment by dave-mitchell** [comment]
I, for one, pictured a workflow like:

1. Trace is scored in the wild (in app/testing)
2. API harvests scores and moves into a queue

**Comment by droos1** [comment]
I think it woudl be great. And even better being able to set a target ike with llm judge, so when you do a dataset run, they are automatically added to the annotation quueu.

**Comment by laisalvescw** [comment]
I would greatly benefit from a feature that allows adding traces to a Human Annotation queue. Specifically, I need a way to send selected trace IDs directly to the queue.

Currently, my workflow involves extracting traces using the Python SDK, running a custom evaluation pipeline, sending them back to Langfuse, and then manually filtering for annotation. However, I’d like to streamline this process by automatically filtering traces within my external pipeline and sending selected (filtered) traces IDs to a new queue on a weekly basis. This would significantly improve efficiency and reduce manual effort.

**Comment by tangochris** [comment]
I'd like to be able to set criteria to automatically add to annotation queue. For example, if "hallucinate" evaluator scores above 0.7, send for human review. Would rather do this from UI than API.

**Reply by marcklingen** [reply]
thanks for sharing, this aligns well with the workflow feature explained here: https://github.com/orgs/langfuse/discussions/5007#discussioncomment-12082564

**Comment by jannikmaierhoefer** [comment]
This feature has been added [here](https://langfuse.com/changelog/2025-03-13-public-api-annotation-queues).

**Comment by dpluscec-sumer-janea** [comment]
Hi, I'm wondering if you can add an endpoint for annotation queue creation as I'd like to automatically generate annotation queues?
